src/locale/ngage/SDL_syslocale.cpp

```diff
@@ -0,0 +1,300 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2025 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "SDL_internal.h"
+#include "../SDL_syslocale.h"
+
+#ifdef NULL
+#undef NULL
+#endif
+
+#include <e32std.h>
+#include <bautils.h>
+#include <e32base.h>
+#include <e32cons.h>
+
+bool SDL_SYS_GetPreferredLocales(char *buf, size_t buflen)
+{
+    TLanguage language = User::Language();
+    const char* locale;
+
+    switch (language) {
+        case ELangFrench:
```

Comment:
This might be a bit more readable as a static table rather than a huge switch statement.

Reply:
As it is an enum, this was the most obvious choice. I can change it if necessary.